server/src/main/java/org/apache/iotdb/db/consensus/ConsensusMain.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.consensus;
+
+import org.apache.iotdb.consensus.IConsensus;
+import org.apache.iotdb.consensus.common.ConsensusGroupId;
+import org.apache.iotdb.consensus.common.Endpoint;
+import org.apache.iotdb.consensus.common.GroupType;
+import org.apache.iotdb.consensus.common.Peer;
+import org.apache.iotdb.consensus.standalone.StandAloneConsensus;
+import org.apache.iotdb.consensus.statemachine.EmptyStateMachine;
+import org.apache.iotdb.db.consensus.ratis.RatisDataRegionStateMachine;
+import org.apache.iotdb.db.consensus.ratis.RatisSchemaRegionStateMachine;
+import org.apache.iotdb.db.exception.metadata.IllegalPathException;
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.qp.physical.crud.InsertRowPlan;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+
+import java.util.Collections;
+
+public class ConsensusMain {
+
+  public static void main(String[] args) throws IllegalPathException {
+
+    IConsensus consensusImpl =
+        new StandAloneConsensus(
+            id -> {
+              switch (id.getType()) {
+                case SchemaRegion:
+                  return new RatisSchemaRegionStateMachine();
+                case DataRegion:
+                  return new RatisDataRegionStateMachine();
+              }
+              return new EmptyStateMachine();
```

Comment:
Is there any place where users can set what consensus algorithm to use for DataRegion?

Reply:
This level of flexibility is currently difficult to achieve due to the default multi-raft implementation built into the Ratis framework. In the future when integrating SOFA-JRAFT we will hopefully be able to specify a special consensus algorithm for dataRegion when creating a consensus instance. My current expectation is that all dataregions should only use the same consensus algorithm, if this is the case then this can be resolved in the form of a configuration file.